### PR TITLE
Added a blog post about the new CLI

### PR DESCRIPTION
**Pull request by fabiospampinato**
[Rendered](https://deploy-preview-15731--prettier.netlify.app/blog/2023/11/29/cli-deep-dive)

### CODE REVIEW

File: website/blog/2023-11-29-cli-deep-dive.md [hunk 1]
```diff
@@ -0,0 +1,279 @@
...
+
+## Finding files fast
+
+First of all we need to find our target files. Prettier's current CLI uses [`fast-glob`](https://github.com/mrmlnc/fast-glob) for that, and the code for using it may look like this:
+
+```js
+import fastGlob from "fast-glob";
+
+const files = await fastGlob("packages/**/*.{js,cjs,mjs}", {
+	ignore: ["**/.git", "**/.sl", "**/.svn", "**/.hg", "**/node_mdules"],
```
Comment (uncenter) [inline website/blog/2023-11-29-cli-deep-dive.md:66]:
```suggestion
	ignore: ["**/.git", "**/.sl", "**/.svn", "**/.hg", "**/node_modules"],
```

File: website/blog/2023-11-29-cli-deep-dive.md [hunk 2]
```diff
@@ -0,0 +1,279 @@
...
+
+1. It should be possible in some (most?) cases to pre-resolve these configuration files for any possible file path we could encounter, for example depending on the globs in them it may be possible that we can end up with 3 possible resolved configurations at most, one for files that don't match any of the globs, one for files that match the `**/*.js` glob, and one for files that match the `**/*.md` glob. It's a bit complicated to implement though, and unclear what the speedup will be in practice, but it's something to think about for the future.
+
+#### Resolving Prettier configurations
+
+For Prettier-specific configurations, like the `.prettierrc` file, we are also assuming that we've resolved all the found configuration files, and can retrieve them in constant time for any target file.
+
+It's basically an identical situation to what we had for EditorConfig-specific configurations, so we'll basically do the same, this time hard-coding the logic for merging the configurations inside the CLI itself, as making a standalone package for it seems of basically no utility to the ecosystem.
+
+The main aspects to consider here for the future are, imo:
```
Comment (uncenter) [inline website/blog/2023-11-29-cli-deep-dive.md:153]:
```suggestion
The main aspects to consider here for the future are, in my opinion:
```

File: website/blog/2023-11-29-cli-deep-dive.md [hunk 3]
```diff
@@ -0,0 +1,278 @@
...
+
+```sh
+prettier src --check # Like before, but faster
+```
+
+You can also try it via `npx`, though `npx` itself is pretty slow:
+
+```sh
+npx prettier@next src --check
+```
```
Comment (fisker) [inline website/blog/2023-11-29-cli-deep-dive.md:27]:
````suggestion
```sh
prettier . --check # Like before, but faster
```

You can also try it via `npx`, though `npx` itself is pretty slow:

```sh
npx prettier@next . --check
```
````

**Comment by fisker** [comment]
Can you mention somewhere that `PRETTIER_LEGACY_CLI=1 prettier` can switch back to using the old CLI, if users find some edge case not supported?

**Review by sosukesuzuki — APPROVED** [review]

**Review by fisker — APPROVED** [review]

**Pull request merged**